### ISSUE TITLE
Result Tracking

### ISSUE THREAD

**Issue by ebprihar** [moorea1/DS501_Case2#2]
This issue is for us to post some images of our results, ask questions, and give feedback.

**Comment by ebprihar** [comment]
Here are my results:

![image](https://user-images.githubusercontent.com/15816792/67124841-63fcba00-f1c1-11e9-8bf1-abbe0079ab4e.png)

**Comment by mbarger1** [comment]
I have 10 funky graphs with pairwise comparison results. I pushed them in a folder to the main page because it would be silly to put them all here.

**Comment by alexander-moore** [comment]
Looks great everyone!!!
Tomorrow morning I’ll have some model studies pushed

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by alexander-moore** [comment]
I have the framework for my exploration, where I can take in a folder full of data sets and handle them each individually.

The problem is I don't know what worth exploring: fitting models on some of the columns (Min, Max, Closing) and predicting Volume traded that week?

Maybe predicting next weeks Min, Max Closing given the historical data?
(Then looking at the coefficient certainties to make conclusions about the relationship of predictor to label)

Not sure what's interesting. If anyone was inspiration or a guiding idea I can implement and post it

**Comment by mbarger1** [comment]
Honestly, both of those ideas sound pretty good. If it isn't too difficult, feel free to do both. If it's tricky, I'd go for predict the volume traded because a naive method for predicting price won't give great results.

**Comment by alexander-moore** [comment]
![heatmap_full](https://user-images.githubusercontent.com/31463017/67167700-00ae8b80-f36b-11e9-813c-e0e2a8b4c8f4.png)

This is a heatmap of correlation in weekly % gain in closing price for the 30 DOW. Now I'll find which have the lowest variance and highest mean

![q_stat](https://user-images.githubusercontent.com/31463017/67168092-87b13300-f36e-11e9-9251-790d23ed2878.png)

these are the highest mean percent increase and the lowest standard deviation of percent increase for the last 10 years.
In Stonks, you want something with high mean increase (obviously) and low volatility. I'm sure this is a common measure but we introduce the Quincy Stat ^TM:
mean / sd

To find 'HD' , The Home Depot, to maximize this statistic. Comment your concerns or suggestions (or compliments)!

**Comment by mbarger1** [comment]
I'll have more results (taking the Dow Inc info out of consideration) tomorrow morning! It's taking longer than I thought to put together.

**Comment by qh2150** [comment]
Still working on it guys.  Hey Mia, do you have the code you used to pull the pricing?

**Comment by alexander-moore** [comment]
https://finance.yahoo.com/quote/DIS/history?p=DIS

We pulled the pricing by manually going to the companies on yelp, clicking "historical pricing" and selecting all available data, weekly. This is whats in the 'data' folder you can download from this git. If you need monthly instead of weekly I'd just do the mean of each 4 rows instead of re-downloading by hand

**Comment by qh2150** [comment]
Ok, no worries, I was just curious if anyone had used the quantopian pipeline interface.  Do we care about the source?

**Comment by alexander-moore** [comment]
Ethan has code in ethan.ipnb where he draws from quantopian for gold prices. There's also samples on their website I saw. 
We don't care about source, just whichever is most convenient for the goals of the sub-project I'd say

**Comment by qh2150** [comment]
Ok, I'm going to mess with quantopian a bit more tonight and then hopefully have something pulled together tomorrow.

**Comment by qh2150** [comment]
Does anyone have a simple way to rotate through the list of DJIA csv's and just pull each closing price column named as the ticker?  If not, I can just pull it off of a terminal at work but it would probably be better if we could combine the dataset into one CSV like that.

**Comment by qh2150** [comment]
I had to temporarily throw in the towel on figuring out the quantopian system and just pulled pricing data out of my Bloomberg Terminal Access for the sake of us actually having something to present.  I uploaded the CSV's which should be easier for you guys to work with one is monthly and one is weekly.  There's a snippet of two lines of code in the comments that pulls it into a dataframe (assumes same folder) and drops V and DOW, since V price data stops prior to 2008 and DOW also is incomplete.  

These datasets only go back to 2000, I can go back further if need be pretty easily later, I was just trying to get us on the Quantopian platform but its really confusing to me.  Right now my goal is to wrap up some of the things we talked about by class.  If you guys want, feel free to play with these datasets, Alexander is it possible to rerun your correlation matrix on both of these datasets?

**Comment by qh2150** [comment]
import numpy as np
import pandas as pd

#Pulls in monthly and weekly DOW30 stock prices back to 2000 from CSV files, drops columns with incomplete price history
data_weekly = pd.read_csv('data_weekly.csv')
data_weekly.drop(['V', 'DOW'], axis=1, inplace=True)

data_monthly = pd.read_csv('data_monthly.csv')
data_monthly.drop(['V', 'DOW'], axis=1, inplace=True)

#data_monthly.head()

**Comment by qh2150** [comment]
I'm working with the optimization, I updated the two CSV files and here's some simple code to work with, log is analogous to returns:

import numpy as np
import pandas as pd
import datetime as dt

#Pulls in monthly and weekly DOW30 stock prices back to 2000 from CSV files, drops columns with incomplete price history

data_weekly = pd.read_csv('data_weekly.csv')
data_weekly.drop(['V', 'DOW'], axis=1, inplace=True)
data_weekly.Date = pd.to_datetime(data_weekly.Date)
data_weekly.set_index('Date', inplace=True)

data_monthly = pd.read_csv('data_monthly.csv')
data_monthly.drop(['V', 'DOW'], axis=1, inplace=True)
data_monthly.Date = pd.to_datetime(data_monthly.Date)
data_monthly.set_index('Date', inplace=True)

log_weekly = np.log(data_weekly/data_weekly.shift(1))
log_weekly = log_weekly.iloc[1:]

log_monthly = np.log(data_monthly/data_monthly.shift(1))
log_monthly = log_monthly.iloc[1:]

#log_monthly.head()
print(len(log_weekly.row))

**Comment by qh2150** [comment]
![image](https://user-images.githubusercontent.com/54342838/67310991-bfce8800-f4cc-11e9-8902-a754f47486f2.png)

**Comment by qh2150** [comment]
Making progress

**Comment by qh2150** [comment]
That was 5 random stock and this is 30, so I think the next step may be trying to select different combinations of like 5 using some form of methodology.

![image](https://user-images.githubusercontent.com/54342838/67311183-1b007a80-f4cd-11e9-94a6-d811f3b3722d.png)

**Comment by mbarger1** [comment]
My new comparison graphs are up! I excluded the DOW stocks. I used the different individual stock CSVs because wrangling my code to work for the aggregate weekly and monthly data would take me a while--I'm going to work on that more tonight and tomorrow, but I figured it's better to post my results sooner rather than later.

**Comment by mbarger1** [comment]
Quincy, is the information in the monthly and weekly data the opening or closing price for that month? Wait never mind I see the dates

**Comment by alexander-moore** [comment]
I pushed a version of the moving window code. It has samples for how the Chi matrix is made, how the Corr matrix is made, and how we can make a list of how market correlations change through time (according to different window sizss)

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by mbarger1** [comment]
New results are up

**Comment by qh2150** [comment]
I put up a rough draft of the power point (if you guys hate the humor I can remove it) I am still going to incorporate Mia's work, clean up some of my plots and I haven't incorporated the latest stuff from Alex yet.  I am also going to continue inserting some more changes to the jupyter notebook (the one called case2_Main).  I've done a lot of work to the stuff off of the blog at this point, so we can safely call it our own version, (everyone stole it off Markowitz , 1952 anyways).  I may ultimately try doing something in Quantopian later but I feel like we have a fair amount of content as it is at this point and are pretty close to having a final product.  If anyone wants to take over part of the presenting just speak up and I can help you with any parts you aren't sure on.

**Comment by ebprihar** [comment]
I added two brief slides that go over my work in ethanslides.pptx

**Comment by qh2150** [comment]
Is there any sort of data that summarizes the performance from the neural network approach? 

I may do a quick alteration on the consistent gain approach by splitting it across a  training and test set if that's ok.

**Comment by qh2150** [comment]
Are you guys ok with a quick meeting around noon tomorrow to review the final slides and any changes for like 10-15 minutes before we submit them?

Ethan and Mia, would you prefer all of your slides be included or if some later get dropped in the interest of space and time is that ok?

**Comment by mbarger1** [comment]
I've pushed 2 slides with my stuff. If they don't end up making it in there, it's no big deal. And yeah, tomorrow at 12 works!

**Comment by qh2150** [comment]
Ok, well if tomorrow around noon works, I'll plan on meeting whoever can make it, 3rd floor of fuller just to lock down any details.

**Comment by alexander-moore** [comment]
Ill be there tomorrow at noon

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
